### PR TITLE
added initial database fixtures

### PR DESCRIPTION
**Pull request by qusvv**
I added two mechanisms to load data into an initial (empty) database (see INSTALL.md)
1. From a single .json file
```
(venv)$ python3 saskatoon/manage.py migrate --skip-checks
(venv)$ python3 saskatoon/manage.py loaddata saskatoon/fixtures/saskatoon.json
``` 
2. init script (includes initial migration)
```
(venv)$ saskatoon/fixtures/init
```

Method 2 has the advantage that .json files are split into separate, human readable .json files that can be more easily audited/modified before loading (e.g. manually adding cities, states, neighborhoods, etc..)

Note: a super user associated with info@lesfruitsdefendus.org is automatically created

### CODE REVIEW

File: saskatoon/fixtures/member-authuser.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[
+{
+  "model": "member.authuser",
+  "pk": 2,
+  "fields": {
+    "password": "pbkdf2_sha256$260000$B4izrSL5cMDhpz76Ml8oAd$y+vALyJB3hKoOGYietAefzmZQMA0TSPrbGq4WLI7yvs=",
```
Comment (tristanlatr) [inline saskatoon/fixtures/member-authuser.json:6]: We should make sure the production instance doesn't use the same password as the one this sha256 represents.

**Review by tristanlatr — COMMENTED** [review]
Looks good but I can't make it work on my device:

```
(ve) tristan@Tristans-MBP saskatoon-ng % python3 saskatoon/manage.py migrate --skip-checks
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, harvest, member, sessions
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying auth.0012_alter_user_first_name_max_length... OK
  Applying member.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying harvest.0001_initial... OK
  Applying harvest.0002_auto_20210525_1847... OK
  Applying sessions.0001_initial... OK
(ve) tristan@Tristans-MBP saskatoon-ng % python3 saskatoon/manage.py loaddata saskatoon/fixture/saskatoon.json
CommandError: No fixture named 'saskatoon' found.
```

I've also tried the `init` command, but no luck :/

```
(ve) tristan@Tristans-MBP saskatoon-ng % saskatoon/fixtures/init
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, harvest, member, sessions
Running migrations:
  No migrations to apply.
Loading member-city
Installed 4 object(s) from 1 fixture(s)
Loading member-neighborhood
Installed 37 object(s) from 1 fixture(s)
Loading member-state
Installed 1 object(s) from 1 fixture(s)
Loading member-country
Installed 1 object(s) from 1 fixture(s)
Loading member-language
Installed 3 object(s) from 1 fixture(s)
Loading member-actor
Installed 3 object(s) from 1 fixture(s)
Loading member-person
Installed 2 object(s) from 1 fixture(s)
Loading member-organization
Installed 1 object(s) from 1 fixture(s)
Loading member-authuser
Installed 1 object(s) from 1 fixture(s)
Loading harvest-property
Traceback (most recent call last):
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 27, in _decorator
    return method(self, *args, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 103, in delete_pattern
    return self.client.delete_pattern(*args, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/client/default.py", line 304, in delete_pattern
    raise ConnectionInterrupted(connection=client) from e
django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/harvest-property.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "../manage.py", line 21, in <module>
    main()
  File "../manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 78, in handle
    self.loaddata(fixture_labels)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 123, in loaddata
    self.load_label(fixture_label)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 190, in load_label
    obj.save(using=self.using)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/serializers/base.py", line 223, in save
    models.Model.save_base(self.object, using=using, raw=True, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/db/models/base.py", line 774, in save_base
    post_save.send(
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
    return [
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
  File "/Users/tristan/Documents/saskatoon-ng/saskatoon/harvest/signals.py", line 6, in clear_cache_property
    cache.delete_pattern("*property*")
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 34, in _decorator
    raise e.__cause__
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/client/default.py", line 299, in delete_pattern
    for key in client.scan_iter(**kwargs):
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 2131, in scan_iter
    cursor, data = self.scan(cursor=cursor, match=match,
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 2112, in scan
    return self.execute_command('SCAN', *pieces)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/harvest-property.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.
Loading harvest-equipmenttype
Installed 13 object(s) from 1 fixture(s)
Loading harvest-treetype
Installed 30 object(s) from 1 fixture(s)
Loading harvest-equipment
Traceback (most recent call last):
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 27, in _decorator
    return method(self, *args, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 103, in delete_pattern
    return self.client.delete_pattern(*args, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/client/default.py", line 304, in delete_pattern
    raise ConnectionInterrupted(connection=client) from e
django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/harvest-equipment.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "../manage.py", line 21, in <module>
    main()
  File "../manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 78, in handle
    self.loaddata(fixture_labels)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 123, in loaddata
    self.load_label(fixture_label)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 190, in load_label
    obj.save(using=self.using)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/serializers/base.py", line 223, in save
    models.Model.save_base(self.object, using=using, raw=True, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/db/models/base.py", line 774, in save_base
    post_save.send(
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
    return [
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
  File "/Users/tristan/Documents/saskatoon-ng/saskatoon/harvest/signals.py", line 15, in clear_cache_equipment
    cache.delete_pattern("*equipment*")
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 34, in _decorator
    raise e.__cause__
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/client/default.py", line 299, in delete_pattern
    for key in client.scan_iter(**kwargs):
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 2131, in scan_iter
    cursor, data = self.scan(cursor=cursor, match=match,
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 2112, in scan
    return self.execute_command('SCAN', *pieces)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/harvest-equipment.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.
Loading harvest-harvest
Traceback (most recent call last):
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 27, in _decorator
    return method(self, *args, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 103, in delete_pattern
    return self.client.delete_pattern(*args, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/client/default.py", line 304, in delete_pattern
    raise ConnectionInterrupted(connection=client) from e
django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/harvest-harvest.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "../manage.py", line 21, in <module>
    main()
  File "../manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 78, in handle
    self.loaddata(fixture_labels)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 123, in loaddata
    self.load_label(fixture_label)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/management/commands/loaddata.py", line 190, in load_label
    obj.save(using=self.using)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/core/serializers/base.py", line 223, in save
    models.Model.save_base(self.object, using=using, raw=True, **kwargs)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/db/models/base.py", line 774, in save_base
    post_save.send(
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
    return [
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
  File "/Users/tristan/Documents/saskatoon-ng/saskatoon/harvest/signals.py", line 9, in clear_cache_harvest
    cache.delete_pattern("*harvest*")
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/cache.py", line 34, in _decorator
    raise e.__cause__
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/django_redis/client/default.py", line 299, in delete_pattern
    for key in client.scan_iter(**kwargs):
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 2131, in scan_iter
    cursor, data = self.scan(cursor=cursor, match=match,
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 2112, in scan
    return self.execute_command('SCAN', *pieces)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/Users/tristan/Documents/saskatoon-ng/ve/lib/python3.8/site-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/harvest-harvest.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.
Unused .json fixtures:
```

The output looks the same for ``saskatoon/fixtures/loaddata all``. 

I've deleted the database file before, so nothing should prevent the fixture to be loaded.

**Comment by qusvv** [comment]
Sorry it's *fixtures* not fixture:
(venv)$ python3 saskatoon/manage.py migrate --skip-check
(venv)$ python3 saskatoon/manage.py loaddata saskatoon/fixtures/saskatoon.json

File: saskatoon/fixtures/member-authuser.json (hunk 1 above)
Comment (qusvv) [reply saskatoon/fixtures/member-authuser.json:6]: Yes for sure! I created a new password that I will send you.
Other thing is if someone else deploys this fixture they must be aware we could log into their admin panel unless they remove this superuser. So maybe having a superuser as initial deployment is not the best idea.
We could also keep this auth.user but remove its superuser status?

Comment (tristanlatr) [reply saskatoon/fixtures/member-authuser.json:6]: Yes, no superuser is a wiser choice.

Comment (qusvv) [reply saskatoon/fixtures/member-authuser.json:6]: Ok I'll fix this tomorrow. Keeping at least one auth.user is necessary to have a pick leader in harvest though. I'll create a new Person called "Pick Leader" as an example and remove Super User

**Comment by qusvv** [comment]
@tristanlatr  Concerning your issue with `saskatoon/fixtures/init` it looks like a redis issue. I ran into something similar today where was unable to create a new harvest or property (`Error 111 connecting to 127.0.0.1:6379`)  unless I ran a `redis-server`

I am not familiar with the use of redis for caching.

**Comment by tristanlatr** [comment]
I can't make is work, I always get 
```
redis.exceptions.ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/saskatoon.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.
```

:/

File: INSTALL.md [hunk 2]
```diff
@@ -161,9 +161,8 @@ $(venv) python3 saskatoon/manage.py createsuperuser --skip-checks
 
 For Django to serve static files during development `SASKATOON_DEBUG` must be set to `yes`.
 
-WARNING: This is not suitable for production use!
+WARNING: This is not suitable for production use! (See [issue 86](https://github.com/LesFruitsDefendus/saskatoon-ng/issues/86))
```
Comment (qusvv) [inline INSTALL.md:164]: To be clear I only meant setting DEBUG=True is not suitable for production. Letting django serve static files in development is not inherently "bad". In production I assume static files are handled by nginx. The main advantage I see of using whitenoise is to get a similar setup in both development and production.

Comment (tiagovaz) [reply INSTALL.md:164]: Hi, just I agree with this change and yes, I always manage static through nginx, e.g

```
    location /static/ {
        alias /var/www/html/saskatoon-ng/saskatoon/sitebase/static/;
    }

```
Bests,

**Comment by qusvv** [comment]
> redis.exceptions.ConnectionError: Problem installing fixture '/Users/tristan/Documents/saskatoon-ng/saskatoon/fixtures/saskatoon.json': Error 61 connecting to 127.0.0.1:6379. Connection refused.

@tristanlatr have you managed to nstall `redis` and run a `redis-server` ? Let's talk about it tomorrow and document that.

Also I had forgotten to load harvest-harverstyield.json in the `init` file but I don't think that caused the issue. (see commit 62dd2497d80a3a4203e5273a542d63dd1832c80a )

File: saskatoon/fixtures/member-authuser.json (hunk 1 above)
Comment (qusvv) [reply saskatoon/fixtures/member-authuser.json:6]: I removed the superuser and only kept one auth.user as pick leader.

**Comment by tiagovaz** [comment]
Hi, I tested your fixtures and worked just fine, thanks! It's missing a few entries (e.g. harvest comments, picker requests) but we can add more data over time. This PR gives a nice start point for fixtures and I think it's ready to be merged.

**Comment by qusvv** [comment]
Hi @tiagovaz, thanks for your feedback! I have added  comment and participation_request fixtures (see commit 496543a8ed77cbebcb2d1570f464b8ffbe60b368) However I have issues with the `send_email` function: ConnectionRefusedError: [Errno 111] Connection refused). So in order  for `saskatoon/fixtures/init` to pass I needed to set `fail_silently=True` in `harvest/signals.py`.

Is there a way to easily setup the email server locally?

**Pull request merged**